functions/Fetch_JSON_Data_from_PublicApi.js

```diff
@@ -0,0 +1,88 @@
+const https = require('https');
+const url = require('url');
+
+exports.handler = (request, context, callback) => {
+    update(request.state, request.secrets, callback);
+};
+
+function update(state, secret, callback) {
+    let result = initializeState(state);
+    let since = state.astroidsCursor || "2018-05-01";
+    let baseApi = "https://api.nasa.gov/neo/rest/v1/feed?start_date=" + new Date(since).toISOString().slice(0,10) + "&api_key=" + secret.apiKey;
+        const request = https.get(baseApi, response => {
+            if (response.statusCode < 200 || response.statusCode > 299) {
+                callback(new Error('Request failed with code: ' + response.statusCode), null);
+            }
+            // temporary data holder
+            let body = "";
+            response.on('data', (chunk) => body += chunk);
+            response.on('end', () => {
+                let insertAsteroids = [], deleteAsteroids = [];
+                let jsonData = JSON.parse(body);
+                let neoData = jsonData["near_earth_objects"];
+                let deleted = false;
+                for (var date in neoData) {
+                    if (neoData.hasOwnProperty(date)) {
+                        if(deleted){
+                            neoData[date].map(a => insertAsteroids.push(flattenOneLevel(a)));
+                        } else {
+                            neoData[date].map(a => deleteAsteroids.push(flattenOneLevel(a)));
+                            deleted = true;
```

Comment:
could you explain the logic here?
why we push to inserts when we met single delete?

Reply:
Actually the idea here is to push data to both `insertAsteroids` and `deleteAsteroids`. Looks like the variable `deleted` used in wrong condition is misleading.